### PR TITLE
Add docs for monitoring API changes in Alpha4

### PR DESCRIPTION
**Pull request by dedemorton**
@jordansissel @andrewvc @suyograo Not sure who needs to review this, but here you go! Please read and respond to the questions that I've added as comments flagged for REVIEWERS. There were a few places where the API calls didn't behave exactly as described in https://github.com/elastic/logstash/issues/5475.

This adds changes for monitoring APIs (tracked by doc issue #5148).

### CODE REVIEW

File: docs/static/monitoring-apis.asciidoc [hunk 1]
```diff
@@ -201,6 +314,117 @@ Example response:
...
+The following request returns a JSON document containing pipeline stats, including the number of events that were
+input, filtered, or output by the pipeline. The request also returns stats for each configured input, filter, or
+output stage. coming[5.0.0-alpha4,Stats for input stages are not yet available]
+
+[source,js]
+--------------------------------------------------
+GET /_node/stats/pipeline
+--------------------------------------------------
+
+//REVIEWERS: Note that when you call /_node/stats, it does not return the pipeline stats. You need to specify /_node/stats/pipeline specifically. This makes me wonder if it shouldn't be its own API topic. WDYT?
```
Comment (suyograo) [inline docs/static/monitoring-apis.asciidoc]: This is intentional. `/_node/stats/pipeline` is too verbose (customers have complex plugin configs) to be included in ` /_node/stats`. Yes it should be a separate API topic.

File: docs/static/monitoring-apis.asciidoc [hunk 2]
```diff
@@ -220,47 +444,78 @@ GET /_node/hot_threads
...
+          "rubyjit.Module$$stoppable_sleep_c19c1639527ca7d373b5093f339d26538f1c21ef1028566121.__file__(/Users/dedemorton/BuildTesting/5.0.0-alpha4/logstash-5.0.0-alpha4/vendor/bundle/jruby/1.9/gems/stud-0.0.22/lib/stud/interval.rb)",
+          "org.jruby.ast.executable.AbstractScript.__file__(AbstractScript.java:46)"
+        ]
+      }
+    ]
+  }
+}
 --------------------------------------------------
 
+//REVIEWERS: Curious why the hot threads API is the only one that (appears) to accept parameters. Am I missing parameters in the other APIs?
```
Comment (suyograo) [inline docs/static/monitoring-apis.asciidoc]: This is the only API so far that accepts parameters.

File: docs/static/monitoring-apis.asciidoc [hunk 3]
```diff
@@ -220,47 +444,78 @@ GET /_node/hot_threads
...
+`ignore_idle_threads`:: If true, does not return idle threads. The default is true.
+
 You can use the `?human` parameter to return the document in a human-readable format.
 
 [source,js]
 --------------------------------------------------
 GET /_node/hot_threads?human=true
 --------------------------------------------------
 
+//REVIEWERS: Does the ?human parameter still work? It didn't work in the build that I used.
```
Comment (suyograo) [inline docs/static/monitoring-apis.asciidoc]: It should, otherwise its a bug.

File: docs/static/monitoring-apis.asciidoc [hunk 4]
```diff
@@ -114,28 +224,27 @@ The node stats API retrieves runtime stats about Logstash.
...
+Gets JVM stats, including stats about threads. added[5.0.0-alpha3,Adds thread count]
 `process`::
-Gets process stats, including stats about file descriptors, memory consumption, and CPU usage. coming[5.0.0-alpha3]   
+Gets process stats, including stats about file descriptors, memory consumption, and CPU usage. added[5.0.0-alpha3] 
+`mem`::
+Gets memory usage stats. coming[5.0.0-alpha4] 
+`pipeline`::
+Gets stats for a pipeline, including stats about each configured input, filter, or output stage. coming[5.0.0-alpha4,Stats for input stages are not yet available] 
+
+//REVIEWERS: In the latest Logstash build available on June 24, the APIs did not behave exactly as they are documented in https://github.com/elastic/logstash/issues/5475. /_node/stats/pipeline returns the expected results, but I couldn't specify anything below that (such as /_node/stats/pipeline/events). So I've documented the end points accordingly. I also noticed that mem appears in the returned JSON doc. Can you help me by suggesting a better/more complete description>?
```
Comment (suyograo) [inline docs/static/monitoring-apis.asciidoc]: This is correct. you cannot do `/_node/stats/pipeline/events` now. Only supported option is `/_node/stats/pipeline`

File: docs/static/monitoring-apis.asciidoc [hunk 5]
```diff
@@ -65,6 +65,110 @@ being consumed by a monitoring tool, rather than intended for human
...
+[source,js]
+--------------------------------------------------
+GET /_node/<types>
+--------------------------------------------------
+
+Where `<types>` is optional and specifies the types of node info you want to return.
+
+You can limit the info that's returned by combining any of the following types: 
+
+//REVIEWERS: It looks like you cannot combine types in a comma-spearated list like you can for the Elasticsearch cluster APIs. Is this intentional/correct behavior? If so, I need to change this to say "<type>".
```
Comment (suyograo) [inline docs/static/monitoring-apis.asciidoc]: This is a bug. We need to support comma separated list like ES. Can we change it as-is for now, and when the bug is fixed we can update it?

**Comment by suyograo** [comment]
@dedemorton I added some comments

File: docs/static/monitoring-apis.asciidoc (hunk 1 above)
Comment (dedemorton) [inline docs/static/monitoring-apis.asciidoc]: Fixed. Made this a separate topic called Pipeline Info API.

File: docs/static/monitoring-apis.asciidoc (hunk 3 above)
Comment (dedemorton) [inline docs/static/monitoring-apis.asciidoc]: @suyograo I created an https://github.com/elastic/logstash/issues/5563 that describes the behavior that I'm seeing in Alpha4 (which is different than Alpha3). If the behavior is intentional, please close the issue. It just doesn't seem quite right to me.

File: docs/static/monitoring-apis.asciidoc (hunk 5 above)
Comment (dedemorton) [inline docs/static/monitoring-apis.asciidoc]: @suyograo I noticed this problem on a couple of APIs. Based on your response, I'm assuming this is a known bug. Let me know if I need to open an issue to track this.

**Comment by dedemorton** [comment]
@suyograo I've made the changes that you suggested and retested the REST calls. Let me know if this is ready to merge.

**Comment by suyograo** [comment]
LGTM

**Pull request closed without merging**